sdb/commands/pretty_print.py

```diff
@@ -29,29 +29,32 @@ class PrettyPrint(sdb.Command):
     names = ["pretty_print", "pp"]
 
     def call(self, objs: Iterable[drgn.Object]) -> None:  # type: ignore
-        baked = [(self.prog.type(type_), class_)
-                 for type_, class_ in sdb.PrettyPrinter.all_printers.items()]
-        handlingClass = None
-        inputType = None
-        firstObj = next(iter(objs), None)
-        if firstObj:
-            inputType = firstObj.type_
-            for type_, class_ in baked:
-                if type_ == inputType and hasattr(class_, "pretty_print"):
-                    handlingClass = class_
-                    break
-
-        if not handlingClass:
-            print("The following types have pretty-printers:")
-            print("\t%-20s %-20s" % ("PRINTER", "TYPE"))
-            for type_, class_ in baked:
-                if hasattr(class_, "pretty_print"):
-                    print("\t%-20s %-20s" % (class_.names[0], type_))
-            if inputType:
-                msg = 'could not find pretty-printer for type {}'.format(
-                    inputType)
+        baked = {
+            sdb.type_canonicalize_name(self.prog, type_): class_
+            for type_, class_ in sdb.PrettyPrinter.all_printers.items()
+        }
+
+        handling_class = None
+        input_type = None
+        first_obj = next(iter(objs), None)
```

Comment:
```
first_obj = next(iter(objs), None)
...
itertools.chain([first_obj], objs)
```
We do the above pretty often...do you think it may be a good idea to create a helper for it?
Maybe the fact that it'd be 3 lines long may not worth it - just wanted to bring it up.

Reply:
Yes, I do this in https://github.com/delphix/sdb/pull/100, see `get_first_type()`

Reply:
I've merged #100, and this PR now takes advantage of get_first_type().  (However, note that we need #103 to exercise this; I've included that commit in this PR too.)